### PR TITLE
Create study_group app

### LINKED ISSUE #41: Missing Study Group app
an application for Study Group functionallity needs to be created, which includes:
* StudyGroup, GroupMember entities need to be created 
*  Data model methods for said entities
* StudyGroupManager  for more complex queries
* Migration files and Model and app registration

### PR DESCRIPTION
**Pull request by idan1356**
- start study_group app
- create models StudyGroup, StudyGroupManager, GroupMember with their respective methods according to data model
- register models to admin and install app in settings.py
- generate initial migration files

Resolves #41 
depends on #45

### CODE REVIEW

**Review by shacharlevy — APPROVED** [review]
LGTM!

**Review by shirazyomtov — APPROVED** [review]
LGTM

File: study_group/models.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def get_all_group_members(self):
+        return self.group_members.all()
+
+    # create a study group and add creator to group
+    @classmethod
+    def create(cls, **kwargs):
+        owner_id = kwargs.pop('owner_id', None)
+        # also need to check if owner_id exists in user table
+        if owner_id is None:
+            raise ValueError("Group owner doesn't exist")
```
Comment (natifridman) [inline study_group/models.py:38]: Doesn't this part of the user relationship?

Comment (idan1356) [reply study_group/models.py:38]: I figured its best to enforce adding owner to group in the creation method, given the owner ID. 
Waiting for the User table before I can implement it properly

**Comment by galitf** [comment]
Please resolve the conflict

**Pull request closed without merging**